docs/SetupTestEnvironment.md

```diff
@@ -0,0 +1,38 @@
+# Setup Test Environment
+
+## Create a Service Principal
+
+- Create a new Service Principal on Azure using [azure cli](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?toc=%2Fen-us%2Fazure%2Fazure-resource-manager%2Ftoc.json&bc=%2Fen-us%2Fazure%2Fbread%2Ftoc.json&view=azure-cli-latest) or [azure portal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal?view=azure-cli-latest)
+- Assign the new Service Principal with [Contributor Role]
+(https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals#assign-role-to-service-principal) so it can access resources on your subscription. This Service Principal will be assigned to Kubernetes to enable it to create and manage its own resources, as well as push build artifacts into Azure Container Registry.
+
+## Create an Azure Container Registry
+
+- Create a new [Azure Container Registry](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal#create-a-container-registry)
+
```

Comment:
Before we create a new AKS cluster, we need to setup Log Analytics that the AKS cluster will use. We dont want the AKS creation to create one because it puts it in a weird resource group

Reply:
I tried creating a log analytics manually and setting the rg. it didnt allow it, thats why we left it as is.